waltz-tools/src/test/java/com/wepay/waltz/tools/server/ServerCliTest.java

```diff
@@ -116,28 +120,28 @@ public void testAddAndRemovePreferredPartition() throws Exception {
             String[] args1 = {
                 "add-preferred-partition",
                 "--server", serverEndpoint1,
-                "--partition", String.valueOf(partitionId),
+                "--partition", partitionIdInterval,
                 "--cli-config-path", configFilePath
             };
             ServerCli.testMain(args1);
 
-            // Verify preferred partition P3 is added to the server descriptor of server 1
+            // Verify preferred partitions P3,P6,P7 are added to the server descriptor of server 1
             serverDescriptors = helper.getWaltzServerRunner().getServerDescriptors();
             for (ServerDescriptor serverDescriptor : serverDescriptors) {
                 if (serverDescriptor.endpoint.toString().equals(serverEndpoint1)) {
-                    assertTrue(serverDescriptor.partitions.contains(partitionId));
+                    partitionIds.forEach(partitionId -> assertTrue(serverDescriptor.partitions.contains(partitionId)));
```

Comment:
Instead of checking for one partition at a time, you can just do `assertTrue(serverDescriptor.partitions.containsAll(partitionIds))`